gcc/rust/typecheck/rust-hir-inherent-impl-check.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2020-2025 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+#ifndef RUST_HIR_INHERENT_IMPL_ITEM_CHECK_H
+#define RUST_HIR_INHERENT_IMPL_ITEM_CHECK_H
+
+#include "rust-diagnostics.h"
+#include "rust-hir-item.h"
+#include "rust-hir-type-check-base.h"
+#include "rust-mapping-common.h"
+#include "rust-type-util.h"
+
+namespace Rust {
+namespace Resolver {
+
+class PrimitiveImplCheck : public TypeCheckBase
+{
+public:
+  static void go ()
+  {
+    PrimitiveImplCheck pass;
+
+    pass.scan ();
+  }
+
+private:
+  void scan ()
+
+  {
+    std::vector<HIR::ImplBlock *> possible_primitive_impl;
+    mappings.iterate_impl_blocks ([&] (HirId id, HIR::ImplBlock *impl) -> bool {
+      // filtering  trait-impl-blocks
+      if (impl->has_trait_ref ())
+	return true;
+      HirId impl_ty_id = impl->get_type ().get_mappings ().get_hirid ();
```

Comment:
I think some line breaks here will help make this easier to read maybe after those returns